src/metabase/api/public.clj

```diff
@@ -67,7 +68,7 @@
   (card-with-uuid uuid))
 
 (defn- transform-results [results]
-  (if (= (:status results) :failed)
+  (if (and (= (:status results) :failed) (not (re-matches error-whitelist (:error results))))
```

Comment:
I don't know what the reason we don't just return the original error message in the first place is, always returning `"An error occurred while running the query."` is not helpful. We should probably just do that instead

Reply:
Oh wait I just realized this is for public queries. I think a key in the `ex-info` data map that lets us know the message is ok to show in embedded situtations is preferrable to a whitelist that is guaranteed to get out of sync at some point. Not sure what to call it. `:show-message-in-embeds?` or something is probably workable.

Reply:
Also the whitelist isn't really going to work unless someone is using English

Reply:
Yes, I mentioned this was a temporary solution:

> I went ahead and pushed some temporary fixes but they will break if the error message is i18n translated.